app/src/main/java/com/ifttt/groceryexpress/MainActivity.kt

```diff
@@ -111,28 +106,31 @@ class MainActivity : AppCompatActivity() {
             .setPositiveButton(R.string.login) { _, _ ->
                 val newEmail = emailView.editText!!.text.toString()
                 emailPreferencesHelper.setEmail(newEmail)
-                val configuration = ConnectButton.Configuration.Builder.withConnectionId(
-                    CONNECTION_ID,
-                    newEmail,
-                    credentialsProvider
-                    , REDIRECT_URI
-                ).setOnFetchCompleteListener { connection ->
-                    findViewById<TextView>(R.id.connection_title).text = connection.name
-                }.build()
+                val configuration = ConnectButton.Configuration.newBuilder(
+                    newEmail, REDIRECT_URI
+                ).withConnectionId(CONNECTION_ID)
+                    /*
+                    Simulating user login. In this case, we want to give ConnectButton a new ConnectionApiClient, so
+                    that it doesn't use the default, global ConnectionApiClient that has the previously set user token.
+                     */
```

Comment:
Is this still required, now that we are going to fetch a user token before an api call?

Reply:
Yes, this is because we are also caching it in the TokenInterceptor. We can alternatively just don't cache it and always call the method. I think this will make the token management better, and we will leave the decision of caching to the application.

Reply:
However, we do have a use case of checking whether the user token is already present, in order for ConnectButton to skip the email field. We will need to figure something out to continue supporting this functionality.